index.js

```diff
@@ -119,17 +120,15 @@ function resolveValue(value, variables, source) {
 module.exports = function(options) {
   return function(style) {
     options = options || {}
-    var userVariables = options.variables || {}
-    var variables =
-      Object.keys(userVariables)
-        .reduce(function(acc, key) {
-          if (key.indexOf("--") !== 0) {
-            acc["--" + key] = userVariables[key]
-          }
-          acc[key] = userVariables[key]
-          return acc
-        }, {})
-    var preserve = (options.preserve === true ? true : false)
+    var variables = options.variables || {}
+    Object.keys(variables).forEach(function(name) {
+      if (name.slice(0, 2) !== "--") {
+        variables["--" + name] = variables[name]
+        delete variables[name]
+      }
+    })
```

Comment:
I will merge this PR, but please next time, do not introduce some mutability when it has been deliberately moved.
I will rewrite this plugin as es6.